scripts/unassign-inactive-issues.mjs

```diff
@@ -0,0 +1,169 @@
+    }
+
+    console.log('Token exists:', !!token);
+
+    const inactivityPeriodInMinutes = 10;
+
+    const repository = process.env.GITHUB_REPOSITORY;
+    if (!repository) {
+      throw new Error('GITHUB_REPOSITORY environment variable is not set');
+    }
+
+    const [owner, repo] = repository.split('/');
+    console.log(`Processing repository: ${owner}/${repo}`);
+
+    const octokit = new Octokit({
+      auth: token,
+      request: {
+        fetch: fetch
+      }
+    });
+
+    try {
+      const authUser = await octokit.rest.users.getAuthenticated();
+      console.log('Successfully authenticated with GitHub as:', authUser.data.login);
+    } catch (authError) {
+      console.error('Authentication error details:', authError);
+      throw new Error(`Authentication failed: ${authError.message}. Please check your token permissions.`);
+    }
+
+    // List open issues
+    const issues = await octokit.rest.issues.listForRepo({
+      owner,
+      repo,
+      state: 'open',
+      per_page: 100,
+    });
+
+    console.log(`Found ${issues.data.length} open issues`);
+
+    for (const issue of issues.data) {
+      const assignee = issue.assignee;
+      if (!assignee || assignee.site_admin) {
+        continue;
+      }
+
+    //   console.log(`\nChecking issue #${issue.number}:`);
+    //   console.log('Issue data:', {
+    //     number: issue.number,
+    //     title: issue.title,
+    //     assignee: assignee.login,
+    //     has_pr_field: !!issue.pull_request,
+    //     updated_at: issue.updated_at
+    //   });
+
+      // Check for linked PRs using multiple methods
+      const checkLinkedPRs = async () => {
+        try {
+          // Method 1: Check direct PR field
+          if (issue.pull_request) {
+            console.log(`Issue #${issue.number} is a pull request`);
+            return true;
+          }
+
+          // Method 2: Check timeline events
+          const timeline = await octokit.rest.issues.listEventsForTimeline({
+            owner,
+            repo,
+            issue_number: issue.number,
+            per_page: 100
+          });
+
+          console.log(`Timeline events for issue #${issue.number}:`, 
+            timeline.data.map(event => ({
+              event: event.event,
+              type: event.event_type,
+              actor: event.actor?.login,
+              created_at: event.created_at
+            }))
+          );
+
+          // Look for various PR-related events
+          const hasPRLink = timeline.data.some(event => 
+            event.event === 'cross-referenced' ||
+            event.event === 'connected' ||
+            event.event === 'referenced' ||
+            (event.source && event.source.type === 'pull_request')
+          );
+
+          if (hasPRLink) {
+            console.log(`Issue #${issue.number} has PR links in timeline`);
+            return true;
+          }
+
+          // Method 3: Check issue body for PR links
```

Comment:
I appreciate you're thoughtful and offer several methods to make this robust.

Regarding specifically 'Method 3', I am not yet entirely sure if we should do this, since in some issues we may have some PRs referenced, but that doesn't necessarily mean it's the 'right PR' resolving that issue. Any thoughts @rtibbles?

Reply:
Thanks for pointing that out @MisRob !, @rtibbles  Should I go ahead and remove Method 3?

Reply:
I'd say likely, but let's wait for @rtibbles's opinion here, perhaps we can think of something else :)

Reply:
I think we can remove the Method 3 for now

Reply:
I think we can keep it after all for now - the regex should limit it to relevant PRs. I don't know why I didn't notice it before, apologies.

Reply:
All good, thanks!